### PR TITLE
Feature/annotation tracking w api

### PR DESCRIPTION
**Pull request by mbwatson**
this PR:

- removes previous annotation counting functionality that was maintained in the application state,
    + and replaces it with an integration with the new API endpoints, `/get_user_annotation_info/<user-id>/`
- renames the annotation-summary component's file to match the component name.

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -21,4 +21,5 @@ export const api = {
   saveAnnotations: '/save_annotations/',
   getImageAnnotations: id => `/get_image_annotations/${ id }/`,
   getAccountDetails: id => axios.get(`get_user_info/${ id }`),
+  getUserAnnotations: id => axios.get(`http://localhost:8000/get_user_annotation_info/${ id }/`),
```
Comment (hyi) [inline src/api.js]: @mbwatson Should ```http://localhost:8000``` be removed from the url in axios.get() above?

Comment (mbwatson) [reply src/api.js]: haha yep!

**Comment by mbwatson** [comment]
@hyi i've rebased ths onto a current `stage` branch, so all those merge conflicts are gone :relieved: .

this just shows a fairly plain annotation summary with the raw numbers to verify things are working.

File: src/components/annotation-summary/annotation-summary.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        <Panel header={ <Fragment>Annotation Summary for <strong>{ annotationName }</strong></Fragment> } key="1">
+          <strong>During this Session:</strong>{` `}
+            <span className="annotation-count">{ currentAnnotations.positive + currentAnnotations.negative }</span> images{` `}
+            ( <span className="annotation-count positive">{ currentAnnotations.positive }</span> positive,{` `}
+              <span className="annotation-count negative">{ currentAnnotations.negative }</span> negative ) <br />
+          <strong>Overall:</strong>{` `}
+            <span className="annotation-count">{ previousAnnotations.positive + previousAnnotations.negative + currentAnnotations.positive + currentAnnotations.negative }</span> images{` `}
+            ( <span className="annotation-count positive">{previousAnnotations.positive }</span> positive,{` `}
+              <span className="annotation-count negative">{previousAnnotations.negative }</span> negative )<br />
+        </Panel>
```
Comment (hyi) [inline src/components/annotation-summary/annotation-summary.js:40]: @mbwatson I see overall number includes both previous and current rounds, so I assume the positive and negative counts below it should include current round as well, something like {previousAnnotations.positive + currentAnnotations.positive}?

**Review by hyi — COMMENTED** [review]
@mbwatson Looks good to me. Just left one comment for your consideration.

File: src/components/annotation-summary/annotation-summary.js (hunk 2 above)
Comment (mbwatson) [reply src/components/annotation-summary/annotation-summary.js:40]: ah, yep. thanks for catching

File: src/components/annotation-summary/annotation-summary.js [hunk 3]
```diff
@@ -32,8 +32,8 @@ export const AnnotationSummary = ({ annotationName }) => {
               previousAnnotations && (
                 <Fragment>
                   <span className="annotation-count">{ previousAnnotations.positive + previousAnnotations.negative + currentAnnotations.positive + currentAnnotations.negative }</span> images{` `}
-                  ( <span className="annotation-count positive">{previousAnnotations.positive }</span> positive,{` `}
-                    <span className="annotation-count negative">{previousAnnotations.negative }</span> negative )<br />
+                  ( <span className="annotation-count positive">{ previousAnnotations.positive + currentAnnotations.positive }</span> positive,{` `}
+                    <span className="annotation-count negative">{ previousAnnotations.negative + currentAnnotations.positive }</span> negative )<br />
```
Comment (hyi) [inline src/components/annotation-summary/annotation-summary.js]: @mbwatson I believe I caught a copy and paste error - it should be ```currentAnnotations.negative``` on line 36

Comment (mbwatson) [reply src/components/annotation-summary/annotation-summary.js]: wow, yep!

new rule: no commits until my morning coffee from now on.

**Review by hyi — APPROVED** [review]
Looks good. @davidborland Can you take a quick look as well and merge it if it looks good to you?

**Comment by davidborland** [comment]
@hyi and @mbwatson, I think it looks good. One question: how do annotations marked irrelevant factor into positive/negative?

**Comment by davidborland** [comment]
One other thing @mbwatson , when I hover over the dropdown header, the mouse cursor changes to a pointer, indicating that I can click to toggle the collapse. However it only actually activates when I click on the text portion of the header. We should either (1) make the collapse toggle on any click in the header (I think this would be preferable) or (2) only change the mouse cursor to a pointer when over the text.

**Comment by hyi** [comment]
@davidborland Annotated images marked as irrelevant are treated as negative.

**Comment by mbwatson** [comment]
it looks like the unexpected header clicking behavior can be solved by upgrading antd to 4.15 and removing the collapsible="header" prop on the Collapse component.

**Comment by mbwatson** [comment]
this last commit changes antd version from ~4.10 to ~4.15.

**Comment by mbwatson** [comment]
continue holding off on this merge.

@davidborland and i are going to discuss which antd version we want tomorrow.

**Pull request merged**